packages/features/timezone-buddy/components/AvailabilitySliderTable.tsx

```diff
@@ -0,0 +1,165 @@
+          const hours = Math.abs(Math.floor(utcOffsetInMinutes / 60));
+          const minutes = Math.abs(utcOffsetInMinutes % 60);
+          const offsetFormatted = `${utcOffsetInMinutes < 0 ? "-" : "+"}${hours
+            .toString()
+            .padStart(2, "0")}:${minutes.toString().padStart(2, "0")}`;
+
+          return (
+            <div className="flex flex-col gap-2 leading-none">
+              <span className="text-default text-sm font-medium">{time}</span>
+              <span className="text-subtle text-xs ">GMT {offsetFormatted}</span>
+            </div>
+          );
+        },
+      },
+      {
+        id: "slider",
+        header: () => {
+          return (
+            <div className="space-between flex items-center space-x-4">
+              <span>{browsingDate.format("DD dddd MMM, YYYY")}</span>
+              <ButtonGroup>
+                <Button
+                  color="minimal"
+                  variant="icon"
+                  StartIcon={ChevronLeftIcon}
+                  onClick={() => setBrowsingDate(browsingDate.subtract(1, "day"))}
+                />
+                <Button
+                  onClick={() => setBrowsingDate(browsingDate.add(1, "day"))}
+                  color="minimal"
+                  StartIcon={ChevronRightIcon}
+                  variant="icon"
+                />
+              </ButtonGroup>
+            </div>
+          );
+        },
+        cell: ({ row }) => {
+          const { timeZone, dateRanges } = row.original;
+          // return <pre>{JSON.stringify(dateRanges, null, 2)}</pre>;
+          return <TimeDial timezone={timeZone} dateRanges={dateRanges} />;
+        },
+      },
+    ];
+
+    return cols;
+  }, [browsingDate]);
+
+  //we must flatten the array of arrays from the useInfiniteQuery hook
+  const flatData = useMemo(() => data?.pages?.flatMap((page) => page.rows) ?? [], [data]) as SliderUser[];
+  const totalDBRowCount = data?.pages?.[0]?.meta?.totalRowCount ?? 0;
+  const totalFetched = flatData.length;
+
+  //called on scroll and possibly on mount to fetch more data as the user scrolls and reaches bottom of table
+  const fetchMoreOnBottomReached = useCallback(
+    (containerRefElement?: HTMLDivElement | null) => {
+      if (containerRefElement) {
+        const { scrollHeight, scrollTop, clientHeight } = containerRefElement;
+        //once the user has scrolled within 300px of the bottom of the table, fetch more data if there is any
+        if (scrollHeight - scrollTop - clientHeight < 300 && !isFetching && totalFetched < totalDBRowCount) {
+          fetchNextPage();
+        }
+      }
+    },
+    [fetchNextPage, isFetching, totalFetched, totalDBRowCount]
+  );
+
+  useEffect(() => {
+    fetchMoreOnBottomReached(tableContainerRef.current);
+  }, [fetchMoreOnBottomReached]);
+
+  return (
+    <TBContext.Provider
+      value={createTimezoneBuddyStore({
+        browsingDate: browsingDate.toDate(),
+      })}>
```

Comment:
As per zustand docs for creating stores where you pass in the value via props. 

We are doing this wrong in other places